### PR TITLE
add Coding and Commit guidelines

### PR DESCRIPTION
**Pull request by rgascons**
Wrote a little file with some guidelines to follow. They're not really new to all of you and I tried to follow the style of the repo (lowercase at the beginning of the title, short commit messages and use of issues). 

Feel free to propose any changes :)

### CODE REVIEW

**Comment by ofpau** [comment]
Nice, thanks Ricard.

The GNOME guidelines suggest that commits may start with a tag to know what part of the project it affects. I don't think this project is big enough to need that, but do you guys think it'd be a good idea?

It's also said in the guidelines that the first line should start with a capital letter. I'm all for that but as you said we're doing the opposite in this project, could you maybe add a line to the README specifying that? The example shows it right but it wouldn't hurt to make it clear.

**Comment by rgascons** [comment]
Updating the commit with the clarification regarding the lowercase letter :) Maybe it's worth it to discuss this point with @casassg before merging.

As per the tag thing... Honestly it proved to be useful in [gnome-games](https://github.com/Kekun/gnome-games/commits?author=rgascons) as the tag provides context for the commit but only if the tags are short (avoid using more than one word) and clear. Tagging is a good exercise for the developers as well, as it helps reduce the scope of the commits and avoids releasing big commits of code that are hard to debug.

**Comment by casassg** [comment]
I'm not sure about git rebase. I remember having this conversation with @dasilvacontin I think the Github PR lifecycle should enable developers to add more or less commits on a branch w/o affecting other people work. As long as it gets merged with a merge commit I don't see any problem with adding more commits. 

Regarding first line I would add that it needs to be a description of an action: Verb in infinitive (add/remove/fix/modify/indent...) + where has this action been applied to (email send method/review hacker interface...) 

Regarding the rest, I agree with everything.

File: HACKING.md
Comment (dasilvacontin) [inline HACKING.md]: Never seen "should start with a lowercase letter" before and I don't see how it makes sense to do so. Sentences start with an uppercase letter.

File: HACKING.md
Comment (dasilvacontin) [inline HACKING.md]: I disagree with this. It's easier to tell what's changed if you just push a commit with the changes. Simpler too. Squashing commits can be done _after_ the review process has finished. Furthermore, squashing / rebasing can now be done automatically by GitHub – no need to introduce extra friction for new contributors.

**Review by dasilvacontin — CHANGES_REQUESTED** [review]
I've always taken this article for reference: https://chris.beams.io/posts/git-commit/

It has a very concise, simple, and great list of rules towards the middle of it:

> Separate subject from body with a blank line
Limit the subject line to 50 characters
Capitalize the subject line
Do not end the subject line with a period
Use the imperative mood in the subject line
Wrap the body at 72 characters
Use the body to explain what and why vs. how

**Comment by dasilvacontin** [comment]
@ofpau I see tagging only useful for automating process. Like auto generating changelogs, etc. Which mostly makes sense for libraries.

For personal projects, I like "tagging" with emojis, because it makes stuff easier to find (it's easier to look for a symbol/image/color than a word), and because it's prettier too.

File: HACKING.md
Comment (rgascons) [reply HACKING.md]: To keep consistency with the current commit style, as mentioned in the first three messages of this PR. I don't mind Uppercase start, but I thought not messing with the habits of the dev team is more valuable. It's not a big change anyways.

File: HACKING.md
Comment (rgascons) [reply HACKING.md]: Care to elaborate on the GitHub rebasing?

What I want to avoid is:

- ffs is fixed now i swear !!!11111
- Fix again
- Fix
- Commit with relevant changes

**Comment by dasilvacontin** [comment]
> There's a button when merging that lets you squash all commits into a single one. Otherwise, you can go and manually squash commits together in the command line. Rebasing does not help with those kinds of commits.

Effin GitHub email not actually replying to the comment thread.

File: HACKING.md
Comment (dasilvacontin) [reply HACKING.md]: There's a button when merging that lets you squash all commits into a
single one. Otherwise, you can go and manually squash commits together in
the command line. Rebasing does not help with those kinds of commits.

When you join commits when using the `git rebase` tool, you call that squashing commits.

Comment (rgascons) [reply HACKING.md]: Awesome, wasn't aware that GitHub can squash the commits now. Removing that line

File: HACKING.md
Comment (ofpau) [reply HACKING.md]: I really like [the article](https://chris.beams.io/posts/git-commit/) @dasilvacontin shared, I usually try to follow that style too. @casassg would you be ok with switching to capitalized subject line?

Comment (casassg) [reply HACKING.md]: I don't think either should be inforced, there are many open source projects that use both styles (see Kubernetes as an example: https://github.com/kubernetes/kubernetes/commits/master?after=cc568f6433f2d90b30ca8e1d51f3597d60affb11+34 ) I think that leaving it up to the author should be the best. I use lower case because it's faster than having to capitalize, but I believe it's more of a personal opinion and style and it shouldn't be inforced. Developers should be able to decide.

Comment (casassg) [reply HACKING.md]: >Separate subject from body with a blank line
>Limit the subject line to 50 characters
>~Capitalize the subject line~
>Do not end the subject line with a period
>Use the imperative mood in the subject line
>Wrap the body at 72 characters
>Use the body to explain what and why vs. how

Agree with the rest

Comment (casassg) [reply HACKING.md]: UP

**Review by casassg — APPROVED** [review]
lgtm now

**Comment by casassg** [comment]
@dasilvacontin can you dismiss or redo your review?

**Pull request merged**
